woocommerce/woocommerce-gutenberg-products-block
# Collection Price Filter: use `context` instead of `state`

## What
This PR convert the price filter block to use context instead of state to:
- follow the convention between new filter blocks.
- enable price filter block to react to changes from other blocks, like clearing all filters action in the Active Filters block.

This PR also remove the usage of getter and reintroduce the derived state to get the latest context.

## Why
- By converting to use context, we can support multiple blocks on the same page. Practically, we may not do that, but theoretically, it's still better for price filter block to not affect or be affected by other price filter blocks on the same page. The data of the block should be isolated to its corresponding instance only.
- The getter is cached so we can't use it to get the latest context, while derived state isn't.


## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Add the `Collection Filters` and `Product Collection` blocks to the page.
2. On the front end, change the price filter.
3. Click the clear all button, see the price filter block reset to initial state.


## Screenshots or screencast



| Before | After |
| ------ | ----- |
|        |       |

## WooCommerce Visibility


Required:


## Checklist

Required:


Conditional:


## Changelog


> Add suggested changelog entry here.
